File: Visual Studio Code/Recursion/Python/recursive_factorial.py
'''
Programmer:     Corey Jenkins
Date:           January 27, 2022
Program Name:   recursion_demo.py
Purpose:        A demonstration on using a recursive factorial in Python.
'''

# Declare the main module
def main():
    number = int(input("Enter a non-negative number:"))
    
    # This is added by me to demonstrate input validation. It is not in the Pseudocode.
    while(number < 0):
        print("You have entered a negative number.")
        number = int(input("Enter a non-negative number:"))
        
    num_factorial = factorial(number)
    
    print("The factorial of", number, "is", num_factorial)
    
def factorial(n):
    if(n == 0):
        return 1
    else:
        return n * factorial(n - 1)
main()

File: Logic & Design/Visual Studio Community Edition/Recursion/C++/RecursiveFactorial/RecursiveFactorial.cpp
/*
	Programmer:		Corey Jenkins
	Date:			April 29, 2017
	Filename:		RecursionDemo
	Purpose:		A demonstration of recursion.
*/

#include <iostream>
#include <string>
using namespace std;

// We want the factorial function to be made after the main function.
int factorial(int n);

int main()
{
	int number;

	cout << "Enter a non-negative number" << endl;
	cin >> number;
	
	// This is added by me and not in the pseudocode document. Input Validation.
	while (number < 0)
	{
		cout << "You have entered a negative number!" << endl;
		cout << "Enter a non-negative number:" << endl;
		cin >> number;
	}
	

	cout << "the factorial of " << number << " is " << factorial(number) << "." << endl;

	system("pause");
	return 0;
}

int factorial(int n)
{
	if (n == 0)
	{
		return 1; // This will turn out to be 1.
	}
	else
	{
		return n * factorial(n - 1);
	}
}